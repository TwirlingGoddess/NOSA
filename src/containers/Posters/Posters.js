import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

export class Posters extends Component {
  constructor() {
    super()
    this.state = {
      default: {
        title:'YOU\'RE CRAZY!!!',
        text: '...if you think that the Earth is ROUND!??!',
        id: -1
      },

      messages: [
        {
          title: 'Knock Knock... Who\'s there?',
          text: 'EARTH IS FLAT!!!',
          id: 0
        },
        {
          title: 'BABY... GOT... FLAT!',
          text: 'I like flat earth and I cannot lie',
          id: 1
        },
        {
          title: 'If I could be a frisbee...',
          text: 'I WOULD BE THE EARTH',
          id: 2
        },
        {
          title: 'When I say FLAT, you say EARTH',
          text: 'FLAT!!  EARTH!!  FLAT!!  EARTH!!',
          id: 3
        }
      ],

      currentPrintPic: '',
      hidden: 'sure',
    }
  }

  changeText = () => {
    const updatedDefault = this.state.messages.find((message, index) => message.id > this.state.default.id)
    this.setState({
      default: updatedDefault
    })
    if(this.state.default.id >= 3) {
      this.setState({
        default: {
          title:'YOU\'RE CRAZY!!!',
          text: '...if you think that the Earth is ROUND!??!',
          id: -1
        }
      })
    }
  }

  toggleDebunksDisplay = () => {
    this.setState({
      hidden: 'nope'
    })
  }

  addPicToPrint = (url) => {
    this.setState({
      currentPrintPic: url
    })
  }

  printOut = () => {
    window.print()
  }

  render() {
    const newImages = this.props.debunks.map(image => (
      <img className="add-pic" 
            onClick={() => this.addPicToPrint(image.url)} 
            alt='' 
            src={image.url}/>))
    return(
      <div>
        <div className="DebunksContainer">
          <div className = "TemplateContainer">
            <div className="Posters">
              <h1 className="Joke">{this.state.default.title}</h1>
              <img className="current-pic" alt='' src={this.state.currentPrintPic} />
              <h3>{this.state.default.text}</h3>
            </div>
            <button className="toggle-display" onClick={this.toggleDebunksDisplay}>Show Debunked Images</button>
            <button className="change-text" onClick={this.changeText}>Change Text</button>
            <button className="print-out" onClick={this.printOut}>PRINT</button>
          </div>
          <div className={this.state.hidden}>
            {newImages}
          </div>
        </div>
      </div>
    )
  }
}

export const mapStateToProps = ({debunks}) => ({debunks})

Posters.propTypes = {
  debunks: PropTypes.array,
}

export default connect(mapStateToProps, null)(Posters)
