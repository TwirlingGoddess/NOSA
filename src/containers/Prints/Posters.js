import React from 'react';

export const Posters = () => {
  return(
      <div className="Posters">
        <h1>pic a debunk image for your card</h1>
        <h2>enter editable text</h2>
      </div>
    )
}