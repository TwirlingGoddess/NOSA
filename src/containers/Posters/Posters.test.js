import React from 'react';
import { shallow, mount } from 'enzyme';
import { Posters, mapStateToProps, changeText, toggleDebunksDisplay, addPicToPrint, printOut } from './Posters';
import { addDebunks } from '../../actions/index.js';

describe('Posters container', () => {
  let initial;
  let expected;
  let wrapper;

  describe('Posters component', () => {
    wrapper = shallow(<Posters />);

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should have a default state of default', () => {
      const mockState = {
        title:'YOU\'RE CRAZY!!!',
        text: '...if you think that the Earth is ROUND!??!',
        id: -1
      };
      expect(wrapper.contains(mockState.title)).toEqual(true);
    });

    it('should have a default state of messages', () => {
      expected = [
        {
          title: 'Knock Knock... Who\'s there?',
          text: 'EARTH IS FLAT!!!',
          id: 0
        },
        {
          title: 'BABY... GOT... FLAT!',
          text: 'I like flat earth and I cannot lie',
          id: 1
        },
        {
          title: 'If I could be a frisbee...',
          text: 'I WOULD BE THE EARTH',
          id: 2
        },
        {
          title: 'When I say FLAT, you say EARTH',
          text: 'FLAT!!  EARTH!!  FLAT!!  EARTH!!',
          id: 3
        }
      ];
      initial = wrapper.state('messages')
      expect(initial).toEqual(expected);
    });

    it('should have a default state of currentPicPrint', () => {
      wrapper.debug()
      expected = '';
      initial = wrapper.state('currentPrintPic');
      expect(initial).toEqual(expected);
    });

    it('should have a default state of hidden', () => {
      expected = 'sure';
      initial = wrapper.state('hidden');
      expect(initial).toEqual(expected);
    });

    it('should invoke changeText when change-text button is clicked', () => {
      const changeTextMock = jest.fn()
      wrapper = mount(<Posters changeText={changeTextMock}/>)
      wrapper.find('.change-text').simulate('click')
      expect(changeTextMock).toHaveBeenCalled()
    })

    it('should update the state of default when changeText is invoked', () => {
      initial = {
        title:'YOU\'RE CRAZY!!!',
        text: '...if you think that the Earth is ROUND!??!',
        id: -1
      }
      expected = {
        title: 'Knock Knock... Who\'s there?',
        text: 'EARTH IS FLAT!!!',
        id: 0
      }
      wrapper.setState({ default: initial });
      wrapper.instance().changeText();
      expect(wrapper.state('default')).toEqual(expected)
    })

    it('should invoke toggleDebunksDisplay when the toggle-display button is clicked', () => {
      const mockEvent = jest.fn
      wrapper.find('.toggle-display').simulate('click')
      expect(toggleDebunksDisplay).toHaveBeenCalled()
    })

    it('should update the state of hidden when toggleDebunksDisplay is invoked', () => {
      initial = 'sure';
      expected = 'nope';
      wrapper.setState({ hidden: initial });
      wrapper.instance().toggleDebunksDisplay();
      expect(wrapper.state('hidden')).toEqual(expected);
    })

    it('should invoke addPicToPrint when the current-pic button is clicked', () => {
      const mockEvent = jest.fn
      wrapper.find('.add-pic').simulate('click');
      expect(addPicToPrint).toHaveBeenCalled()
    })

    it('should update the state of hidden when addPicToPrint is invoked', () => {
      initial = ''
      expected = "https://epic.gsfc.nasa.gov/archive/natural/2018/09/16/png/epic_1b_20180916003633.png"
      wrapper.setState({ currentPrintPic: initial })
      wrapper.instance().addPicToPrint()
      expect(wrapper.state('currentPrintPic')).toEqual(expected)
    })

    it('should call addPicToPrint with the correct params', () => {
      expected = "https://epic.gsfc.nasa.gov/archive/natural/2018/09/16/png/epic_1b_20180916003633.png"
      wrapper.instance().addPicToPrint()
      expect(addPicToPrint).toHaveBeenCalledWith(expected)
    })

    it('should invoke printOut when the print-out button is clicked', () => {
      const mockEvent = jest.fn
      wrapper.instance().printOut(mockEvent);
      expect(printOut).toHaveBeenCalled()
    })

  })

  describe('mapStateToProps', () => { 
    it('should return an object with the debunks array', () => {
      const mockState = {
        debunks: [{url: 'https://epic.gsfc.nasa.gov.png', id: 3, text: ''}],
      }
      const expected = {
        debunks: [{url: 'https://epic.gsfc.nasa.gov.png', id: 3, text: ''}]
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  })

})