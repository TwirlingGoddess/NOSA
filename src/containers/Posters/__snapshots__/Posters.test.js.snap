// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posters container Posters component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Posters />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="DebunksContainer"
>
        <div
                className="TemplateContainer"
        >
                <div
                        className="Posters"
                >
                        <h1
                                className="Joke"
                        >
                                YOU'RE CRAZY!!!
                        </h1>
                        <img
                                alt=""
                                className="current-pic"
                                src=""
                        />
                        <h3>
                                ...if you think that the Earth is ROUND!??!
                        </h3>
                </div>
                <button
                        className="toggle-display"
                        onClick={[Function]}
                >
                        Show Debunked Images
                </button>
                <button
                        className="change-text"
                        onClick={[Function]}
                >
                        Change Text
                </button>
                <button
                        className="print-out"
                        onClick={[Function]}
                >
                        PRINT
                </button>
        </div>
        <div
                className="sure"
        />
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="TemplateContainer"
>
            <div
                        className="Posters"
            >
                        <h1
                                    className="Joke"
                        >
                                    YOU'RE CRAZY!!!
                        </h1>
                        <img
                                    alt=""
                                    className="current-pic"
                                    src=""
                        />
                        <h3>
                                    ...if you think that the Earth is ROUND!??!
                        </h3>
            </div>
            <button
                        className="toggle-display"
                        onClick={[Function]}
            >
                        Show Debunked Images
            </button>
            <button
                        className="change-text"
                        onClick={[Function]}
            >
                        Change Text
            </button>
            <button
                        className="print-out"
                        onClick={[Function]}
            >
                        PRINT
            </button>
</div>,
          <div
            className="sure"
/>,
        ],
        "className": "DebunksContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="Posters"
>
                <h1
                                className="Joke"
                >
                                YOU'RE CRAZY!!!
                </h1>
                <img
                                alt=""
                                className="current-pic"
                                src=""
                />
                <h3>
                                ...if you think that the Earth is ROUND!??!
                </h3>
</div>,
              <button
                className="toggle-display"
                onClick={[Function]}
>
                Show Debunked Images
</button>,
              <button
                className="change-text"
                onClick={[Function]}
>
                Change Text
</button>,
              <button
                className="print-out"
                onClick={[Function]}
>
                PRINT
</button>,
            ],
            "className": "TemplateContainer",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1
                    className="Joke"
>
                    YOU'RE CRAZY!!!
</h1>,
                  <img
                    alt=""
                    className="current-pic"
                    src=""
/>,
                  <h3>
                    ...if you think that the Earth is ROUND!??!
</h3>,
                ],
                "className": "Posters",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "YOU'RE CRAZY!!!",
                    "className": "Joke",
                  },
                  "ref": null,
                  "rendered": "YOU'RE CRAZY!!!",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "className": "current-pic",
                    "src": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "...if you think that the Earth is ROUND!??!",
                  },
                  "ref": null,
                  "rendered": "...if you think that the Earth is ROUND!??!",
                  "type": "h3",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Show Debunked Images",
                "className": "toggle-display",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Show Debunked Images",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Change Text",
                "className": "change-text",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Change Text",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "PRINT",
                "className": "print-out",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "PRINT",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "sure",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="DebunksContainer"
>
          <div
                    className="TemplateContainer"
          >
                    <div
                              className="Posters"
                    >
                              <h1
                                        className="Joke"
                              >
                                        YOU'RE CRAZY!!!
                              </h1>
                              <img
                                        alt=""
                                        className="current-pic"
                                        src=""
                              />
                              <h3>
                                        ...if you think that the Earth is ROUND!??!
                              </h3>
                    </div>
                    <button
                              className="toggle-display"
                              onClick={[Function]}
                    >
                              Show Debunked Images
                    </button>
                    <button
                              className="change-text"
                              onClick={[Function]}
                    >
                              Change Text
                    </button>
                    <button
                              className="print-out"
                              onClick={[Function]}
                    >
                              PRINT
                    </button>
          </div>
          <div
                    className="sure"
          />
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="TemplateContainer"
>
              <div
                            className="Posters"
              >
                            <h1
                                          className="Joke"
                            >
                                          YOU'RE CRAZY!!!
                            </h1>
                            <img
                                          alt=""
                                          className="current-pic"
                                          src=""
                            />
                            <h3>
                                          ...if you think that the Earth is ROUND!??!
                            </h3>
              </div>
              <button
                            className="toggle-display"
                            onClick={[Function]}
              >
                            Show Debunked Images
              </button>
              <button
                            className="change-text"
                            onClick={[Function]}
              >
                            Change Text
              </button>
              <button
                            className="print-out"
                            onClick={[Function]}
              >
                            PRINT
              </button>
</div>,
            <div
              className="sure"
/>,
          ],
          "className": "DebunksContainer",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="Posters"
>
                  <h1
                                    className="Joke"
                  >
                                    YOU'RE CRAZY!!!
                  </h1>
                  <img
                                    alt=""
                                    className="current-pic"
                                    src=""
                  />
                  <h3>
                                    ...if you think that the Earth is ROUND!??!
                  </h3>
</div>,
                <button
                  className="toggle-display"
                  onClick={[Function]}
>
                  Show Debunked Images
</button>,
                <button
                  className="change-text"
                  onClick={[Function]}
>
                  Change Text
</button>,
                <button
                  className="print-out"
                  onClick={[Function]}
>
                  PRINT
</button>,
              ],
              "className": "TemplateContainer",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h1
                      className="Joke"
>
                      YOU'RE CRAZY!!!
</h1>,
                    <img
                      alt=""
                      className="current-pic"
                      src=""
/>,
                    <h3>
                      ...if you think that the Earth is ROUND!??!
</h3>,
                  ],
                  "className": "Posters",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "YOU'RE CRAZY!!!",
                      "className": "Joke",
                    },
                    "ref": null,
                    "rendered": "YOU'RE CRAZY!!!",
                    "type": "h1",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "",
                      "className": "current-pic",
                      "src": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "...if you think that the Earth is ROUND!??!",
                    },
                    "ref": null,
                    "rendered": "...if you think that the Earth is ROUND!??!",
                    "type": "h3",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Show Debunked Images",
                  "className": "toggle-display",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Show Debunked Images",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Change Text",
                  "className": "change-text",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Change Text",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "PRINT",
                  "className": "print-out",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "PRINT",
                "type": "button",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "sure",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
